### PR TITLE
Using attributes to fill manifest Extra property

### PR DESCRIPTION
**Pull request by lock9**
This is a draft while I can't fix the GitHub workflow issue.

### CODE REVIEW

**Comment by shargon** [comment]
Any advance @lock9 ?

**Comment by Tommo-L** [comment]
@shargon @lightszero We need this feature, as neo3 have removed these properties. Sometimes, we want to contact the author of those contracts. I think we can support in our compilers. Could we put more efforts on this PR?

**Comment by lightszero** [comment]
where is  this info will  write to ? abi?

**Comment by shargon** [comment]
Agree too

**Comment by ShawnYun** [comment]
I have updated this pr to the latest complier. And the UT passed. So I think we can convert the draft PR into formal PR.

**Comment by Tommo-L** [comment]
@shargon @lightszero Could help convert this into a formal PR, since @ShawnYun  have tested, all passed.

**Comment by lock9** [comment]
Hello, 
Sorry, I disabled all GitHub notifications. What change you guys want? 
The reason that there is a single and generic property, is that, according to Erik, and I agree, the user may choose to not fill any of this information, so there is no need for "mandatory fields".

**Comment by Tommo-L** [comment]
Thank you @lock9 , it's a good PR.

**Comment by Tommo-L** [comment]
@lock9  If you have time, could you implement it #210 , or after this PR merged, we'll do it later.

File: src/Neo.Compiler.MSIL/Program.cs [hunk 1]
```diff
@@ -278,5 +283,25 @@ public static int Compile(Options options)
 
             return -1;
         }
+
+        private static string BuildExtraAttributes(List<Mono.Collections.Generic.Collection<CustomAttributeArgument>> extraAttributes)
+        {
+            if (extraAttributes.Count == 0)
+            {
+                return null;
```
Comment (shargon) [inline src/Neo.Compiler.MSIL/Program.cs]: It could produce a invalid ABI

**Comment by shargon** [comment]
Maybe we should use constants?

**Comment by Tommo-L** [comment]
> Maybe we should use constants?

？

**Comment by shargon** [comment]
`[ManifestExtra(ManifestExtraAttribute.Author, "Neo")]`

**Comment by shargon** [comment]
Or maybe create different arributes for this
`AuthorAttribute` `EmailAttribute`... etc

**Comment by Tommo-L** [comment]
```c#
[ManifestExtraAttribute("Name", "Demo")]
[ManifestExtraAttribute("Author", "Neo")]
[ManifestExtraAttribute("Email", "dev@neo.org")]
[ManifestExtraAttribute("Description", "This is a contract example")]
```
vs
```c#
NameAttribute("Neo")
AuthorAttribute("Neo")
EmailAttribute("dev@neo.org")
DescriptionAttribute("This is a contract example")
```

 Personally, I prefer the original approach, which is clearer and more organized. 😂

**Comment by shargon** [comment]
> Personally, I prefer the original approach, which is clearer and more organized. 😂

Developers will use it sometimes in lowercase or upercase...

**Comment by shargon** [comment]
What do you think @neo-project/core @lock9 @neo-project/ngd-shanghai @neo-project/ngd-seattle  ?

**Comment by Tommo-L** [comment]
@shargon How about this format:

```c#
@Name("Demo")
@Author("Neo")
@Email("dev@neo.org")
@Description("This is ...")
@ExtraAttribute("Key1", "value1")
public class ContractA : SmartContract
{
     //....
}
```

**Comment by shargon** [comment]
> Or could we read these information from summary directly?

It's more complicated, I prefer in the IL

**Comment by Tommo-L** [comment]
haha...., or let's merge first, we can optimize later. 🤣

**Pull request merged**